commit b4ad71938a14bbea46a59694789676b7653ca04e
Author: Boyuan Yang <073plan@gmail.com>
Date:   Thu Jan 25 22:02:18 2018 +0800

    src: Sound.cc: Fix warning about ISO C++ in Wwrite-strings

diff --git a/src/Sound.cc b/src/Sound.cc
index bb2043b..a4de28c 100644
--- a/src/Sound.cc
+++ b/src/Sound.cc
@@ -1,12 +1,15 @@
 /*
  * Sound.cc
- * Copyright (C) lerosua 2009 <lerosua@gmail.com>
+ * Copyright (C) 2009 lerosua <lerosua@gmail.com>
+ * Copyright (C) 2018 Boyuan Yang <073plan@gmail.com>
  * 
  */
 
 #include "Sound.h"
 #include <string>
+#include <vector>
 #include <glib.h>
+
 using namespace std;
 
 
@@ -77,14 +80,18 @@ CSound::play (SOUND_EVENTS event)
 void CSound::play_file(const char* filename)
 {
 	gchar* argv[3];
-//#ifdef __linux__
-//	argv[0] = "aplay";
-#ifdef  __APPLE__
-	argv[0] = "afplay";
+
+/* storing the program name in std::string */
+#ifdef __APPLE__
+    const std::string const_pname = "afplay";
 #else
-	argv[0] = "aplay";
+    const std::string const_pname = "aplay";
 #endif
-	argv[1] = (gchar*)filename;
+    std::vector<gchar> pname(const_pname.begin(), const_pname.end());
+    pname.push_back('\0');
+    argv[0] = &*pname.begin(); // retrieve non-const gchar *
+
+	argv[1] = (gchar *)filename;
 	argv[2] = NULL;
 	GError* err;
 	GSpawnFlags flas = (GSpawnFlags)(G_SPAWN_SEARCH_PATH |
